pkg/port/iscsit/session.go

```diff
@@ -286,6 +286,22 @@ func (tq *taskQueue) Pop() interface{} {
 	return item
 }
 
+func (tq *taskQueue) Delete(x interface{}) {
+	item := x.(*iscsiTask)
```

Comment:
Just check the cases where x cannot be type casted to iscsiTask

Reply:
Please put some unit test code to prove the logic.

Reply:
We don't have a go test framework in this repository, libiscsi test suite in jiva takes care of the test cases. This has been manually tested by verifying that the memory leak stops after applying this patch.